### PR TITLE
[SPIR-V] Fix opaque pointers support in intel/llvm/llvm-spirv

### PR DESCRIPTION
**Pull request by MrSidims**
It cherry-picks several missing patches from KHR translator

### CODE REVIEW

File: llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp [hunk 1]
```diff
@@ -154,15 +154,8 @@ static cl::opt<bool>
...
-#if SPIRV_ENABLE_OPAQUE_POINTERS
-constexpr static bool SPIRVOpaquePointersDefault = true;
-#else
-constexpr static bool SPIRVOpaquePointersDefault = false;
-#endif
-
 static cl::opt<bool>
-    EmitOpaquePointers("emit-opaque-pointers",
-                       cl::init(SPIRVOpaquePointersDefault),
+    EmitOpaquePointers("emit-opaque-pointers", cl::init(true),
```
Comment (bader) [inline llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp:158]: I think we still need to keep this change until we are ready to switch the whole stack to opaque pointers.
To test the status, it's better to change the CMAKE variable which will turn on opaque pointers for all components.
Here as an example how I tested it - https://github.com/intel/llvm/pull/9828/files#diff-f6c3fac0de0e2bbd79ac93b0a6aabe77d9961c5487347a2170c78ac77770d8afR895.

Comment (MrSidims) [reply llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp]: Restored

**Comment by MrSidims** [comment]
Had to force push to preserve commit order (having in mind to rebase and merge strategy instead of squash and merge)

File: llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp (hunk 1 above)
Comment (bader) [reply llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp]: Great, but I'd like to understand the status of SYCL tests for DPC++ compiler using opaque pointers.
We need to test this in addition to regular CI pre-commit. I'd like to make sure that check-llvm-spirv test suite passes both settings w/ and w/o opaque pointers.

Comment (bader) [reply llvm-spirv/tools/llvm-spirv/llvm-spirv.cpp]: @MrSidims, could you test this configuration, please?

**Comment by bader** [comment]
@intel/dpcpp-esimd-reviewers, could you take a look at this error, please?
```
error: LLVM ERROR: GenXCisaBuilder failed for: <  %gather1364 = call <8 x <8 x float>> @llvm.genx.svm.gather.v8v8f32.v8i1.v8i64(<8 x i1> <i1 true, i1 true, i1 true, i1 true, i1 true, i1 true, i1 true, i1 true>, i32 0, <8 x i64> %ptrtoint1363, <8 x <8 x float>> undef), !VCAlignment !17>: Bad element type for SVM scatter/gather
```
NOTE: requires opaque pointers to reproduce.

**Review by bader — APPROVED** [review]
LGTM.
Please, update the PR title.

**Review by bb-sycl — APPROVED** [review]

**Comment by bb-sycl** [comment]
Fri 21 Jul 2023 12:42:32 AM UTC --- Start to merge the commit into sycl branch. It will take several minutes.

**Comment by bb-sycl** [comment]
Fri 21 Jul 2023 12:46:32 AM UTC --- Merge the branch in this PR to base automatically. Will close the PR later.

**Pull request merged**